packages/amazon-ssm-agent/amazon-ssm-agent.spec

```diff
@@ -59,12 +92,47 @@ go build -ldflags "${GOLDFLAGS}" -o ssm-session-worker \
 gofips build -ldflags "${GOLDFLAGS}" -o fips/ssm-session-worker \
   ./agent/framework/processor/executer/outofproc/sessionworker/main.go
 
+%set_cross_go_flags_static
+
+go build -ldflags "${GOLDFLAGS}" -o static/amazon-ssm-agent \
+  ./core/agent.go ./core/agent_unix.go ./core/agent_parser.go
+
+gofips build -ldflags "${GOLDFLAGS}" -o fips-static/amazon-ssm-agent \
+  ./core/agent.go ./core/agent_unix.go ./core/agent_parser.go
+
+go build -ldflags "${GOLDFLAGS}" -o static/ssm-agent-worker \
+  ./agent/agent.go ./agent/agent_unix.go ./agent/agent_parser.go
+
+gofips build -ldflags "${GOLDFLAGS}" -o fips-static/ssm-agent-worker \
+  ./agent/agent.go ./agent/agent_unix.go ./agent/agent_parser.go
+
+go build -ldflags "${GOLDFLAGS}" -o static/ssm-session-worker \
+  ./agent/framework/processor/executer/outofproc/sessionworker/main.go
+
+gofips build -ldflags "${GOLDFLAGS}" -o fips-static/ssm-session-worker \
+  ./agent/framework/processor/executer/outofproc/sessionworker/main.go
+
 %install
-# Install the SSM agent under 'libexecdir', since it is meant to be used by other programs
+install -D -p -m 0644 %{S:1} %{buildroot}%{_cross_unitdir}/amazon-ssm-agent.service
+
+# Files copied by tmpfiles.d
+install -d %{buildroot}%{_cross_tmpfilesdir}
+install -p -m 0644 %{S:3} %{buildroot}%{_cross_tmpfilesdir}/amazon-ssm-agent-tmpfiles.conf
+
+install -d %{buildroot}%{_cross_factorydir}%{_cross_sysconfdir}/amazon/ssm
+install -m 0644 %{S:2} %{buildroot}%{_cross_factorydir}%{_cross_sysconfdir}/amazon/ssm/amazon-ssm-agent.json
```

Comment:
As mentioned, no need to include a default config file.

Reply:
> As mentioned, no need to include a default config file.

Why not? Does this config file effectively end up just listing what would be the defaults?

Reply:
My reason for including the custom config `amazon-ssm-agent.json` and the tmpfiles.d to install it was to make it as easy as possible for downstream users to change this config as necessary.

But I also understand the case that we do not want to include this in the core-kit itself. I'm fine with either approach.

Reply:
> > As mentioned, no need to include a default config file.
> 
> Why not? Does this config file effectively end up just listing what would be the defaults?

My reasoning was that it wouldn't be possible for downstreams to override this configuration if we provide it through the package, right? RPM will complain if the downstreams provide the same file twice (unless you know a magic trick in RPM to prevent that). What I thought was that we could provide the factory file, with reasonable defaults, and let downstreams decide whether they want to use it with a `tmpfiles.d` file. If they want to provide their own, they can provide their own factory and their own `tmpfiles.d` conf file so that they copy the correct file.

Reply:
>My reasoning was that it wouldn't be possible for downstreams to override this configuration if we provide it through the package, right?

This config file `amazon-ssm-agent.json` is essentially the default configs + our new config for `UseShell`. So if downstreams want override, they can modify this file (or provide their own `amazon-ssm-agent.json`).

>What I thought was that we could provide the factory file

Can you clarify what you mean by "factory file"?

Reply:
Discussed with @arnaldo2792 offline. We are keeping `amazon-ssm-agent.json` in `%{_cross_factorydir}%{_cross_sysconfdir}`. Customers can opt to take this config by adding a tmpfiles.d for that target or create/install their own config.